content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,33 @@
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you find things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you find things *from* the web (information and answers). Here are some examples `TODO: make it clear which these are available and which are future directions`:
+
+- Trying to figure out the best way to call a function? Code search could show you a semi-exhaustive list of call sites. Code intelligence is also being able to see how the code owners of that function themselves have recently been calling that function.
+- Need to debug an issue in code? Code search could help you navigate within the same project to understand it. Code intelligence is also being able to see what changed and why, and trace calls across projects and dependencies/dependents.
+- Want to fix or improve something across your entire codebase? Code search might help you find where needs fixing. Code intelligence is also being able to push up a quick fix everywhere, track its progress, and monitor for regressions.
+- Need to see how your team's API is being used? Code search could show you a list of file locations. Code intelligence is also seeing which other teams are using it and the trend of usage over time.
+
+A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Code hosts are too conflicted to make a good code intelligence platform because they want you to use *their* offerings in each category (and suffer vendor lock-in), but we're committed to Sourcegraph being universal so you can get full code intelligence *and* keep using the best tools in each category.
+
+Today, we're announcing Sourcegraph 4.0, the code intelligence platform for devs to unblock themselves and code better. We've spent 9 years building the underlying foundation of code search to be fast, secure, and scalable across ~10^11 lines of code for the world's most demanding software teams, including `TODO: describe awesome customers`. `TODO: explicitly justify why we are declaring it a code intelligence platform now, rather than in the future`
+
+Sourcegraph 4.0 is available on Cloud (secure, dedicated instances with SOC2 Type 2 compliance), custom deployments, and self-managed.
```

Comment:
4.0 marks a big evolution/departure from Sourcegraph as a self-managed product. Consider adding a sentence or two on Cloud and what this means for our customers (accessibility, ease of use, etc.)

Reply:
```suggestion
Sourcegraph 4.0 is available on Cloud, custom deployments, and self-hosted. If you're new to Sourcegraph,  Sourcegraph Cloud (secure & dedicated instances) is the easiest way to get started. You can get a (30-day trial here)[www.signup.sourcegraph.com].
```